src/main/java/seedu/sudowudo/model/item/Item.java

```diff
@@ -263,42 +256,45 @@ public String toString() {
         return description.toString();
     }
 
-
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof ReadOnlyItem // instanceof handles nulls
-                && this.isSameStateAs((ReadOnlyItem) other));
+                        && this.isSameStateAs((ReadOnlyItem) other));
     }
 
-    //@@author A0147609X
+    // @@author A0147609X
     @Override
     /**
      * sort by start date then end date then alphabetically
      * for UI chronological sort
      * @author darren
      */
     public int compareTo(Item other) {
-        LocalDateTime thisStart, thisEnd, otherStart, otherEnd;
-        
-        thisStart = assignDummyLDT(startDate);
-        thisEnd = assignDummyLDT(endDate);
-        otherStart = assignDummyLDT(other.getStartDate());
-        otherEnd = assignDummyLDT(other.getEndDate());
+        LocalDateTime thisStart = assignDummyLDT(this.startDate);
+        LocalDateTime thisEnd = assignDummyLDT(this.endDate);
+        LocalDateTime otherStart = assignDummyLDT(other.getStartDate());
+        LocalDateTime otherEnd = assignDummyLDT(other.getEndDate());
         
-        if(thisStart.isBefore(otherStart)) {
+        // Assign same start/end date to a deadline for easier checking
+        if (this.is(Type.TASK.toString())){
+            thisStart = thisEnd;
+        }
+        if (other.is(Type.TASK.toString())){
+            otherStart = otherEnd;
+        }
+
+        if(thisEnd.isBefore(otherEnd)) {
+        // this item ends earlier
+            return -1;
+        } else if(thisEnd.isAfter(otherEnd)){
```

Comment:
Coding standard. Add whitespace before brace (just run the auto-formatter on the entire class).